SDL2 fails to build

Here's the error output when building with `x86_64-w64-mingw32.static`. This also happens on `i686-w64-mingw32.static`.
```
Failed to build package sdl2 for target x86_64-w64-mingw32.static!
------------------------------------------------------------
make -C '/home/james/git-repos/mxe/tmp-sdl2-x86_64-w64-mingw32.static/SDL2-2.0.12' -j '4'
make[2]: Entering directory '/home/james/git-repos/mxe/tmp-sdl2-x86_64-w64-mingw32.static/SDL2-2.0.12'
make[2]: *** No rule to make target 'build/SDL.lo', needed by 'build/libSDL2.la'.  Stop.
make[2]: Leaving directory '/home/james/git-repos/mxe/tmp-sdl2-x86_64-w64-mingw32.static/SDL2-2.0.12'
make[1]: *** [Makefile:846: build-only-sdl2_x86_64-w64-mingw32.static] Error 2
make[1]: Leaving directory '/home/james/git-repos/mxe'
real	0m30.685s
user	0m31.345s
sys	0m4.033s
------------------------------------------------------------
[log]      /home/james/git-repos/mxe/log/sdl2_x86_64-w64-mingw32.static
```
I'm not sure if the same happens on the shared targets, but I'll update this if it fails there too.
EDIT: It also fails on shared targets.

[Here's the log files for both x86_64 and i686](https://gist.github.com/TheGreatMcPain/bc5415234d3f36d640257351e6149338)

I'm on Gentoo Linux.

That's interesting - I'm running Gentoo Linux too and I was able to compile SDL2 2.0.12 without issues when it first appeared here. I haven't tried to rebuild it afterwards, so there's the possibility that some seemingly unrelated commit affected this.

I'll try to rebuild it in a fresh clone of the mxe repo to verify/retest, but it will take a while since my build VM is quite slow at the moment.

I'm a bit confused about `make[2]: *** No rule to make target 'build/SDL.lo', needed by 'build/libSDL2.la'.  Stop.` here, since I'd expect something like 'build/SDL2.lo' instead.

--- a/src/sdl2.mk
+++ b/src/sdl2.mk
@@ -18,7 +18,7 @@ define $(PKG)_UPDATE
 endef
 
 define $(PKG)_BUILD
-    cd '$(1)' && aclocal -I acinclude && autoconf && ./configure \
+    cd '$(1)' && aclocal -I acinclude && autoconf && $(SHELL) ./configure \
         $(MXE_CONFIGURE_OPTS) \
         --enable-threads \
         --enable-directx \
@@ -27,8 +27,8 @@ define $(PKG)_BUILD
     $(SED) -i 's,defined(__MINGW64_VERSION_MAJOR),defined(__MINGW64_VERSION_MAJOR) \&\& defined(_WIN64),' '$(1)/include/SDL_cpuinfo.h'
     $(SED) -i 's,-XCClinker,,' '$(1)/sdl2.pc'
     $(SED) -i 's,-XCClinker,,' '$(1)/sdl2-config'
-    $(MAKE) -C '$(1)' -j '$(JOBS)'
-    $(MAKE) -C '$(1)' -j 1 install
+    $(MAKE) -C '$(1)' -j '$(JOBS)' SHELL=$(SHELL)
+    $(MAKE) -C '$(1)' -j 1 install SHELL=$(SHELL)
     ln -sf '$(PREFIX)/$(TARGET)/bin/sdl2-config' '$(PREFIX)/bin/$(TARGET)-sdl2-config'
 
     '$(TARGET)-gcc' \

